src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -42,8 +42,15 @@ private void handleCommandEntered() {
         }
 
         try {
-            commandExecutor.execute(commandText);
+            CommandResult commandResult = commandExecutor.execute(commandText);
+
+            if (commandResult.isPrevious()) {
+                final String previousCommand = commandResult.getFeedbackToUser().split("\\|+")[0];
+                commandTextField.setText(previousCommand);
+                return;
+            }
```

Comment:
I like how the regex with the greedy check is used here to ensure that even with multiple "|" used, the string can still be split correctly. 👍

Reply:
Thank you :D